### DISCUSSION TITLE
B.C and A.C dates

### DISCUSSION THREAD

**Discussion by cgrillo98** [CollectionBuilder/collectionbuilder.github.io#87] [Q&A] [answered]
Hello, does anyone know if it is possible to add B.C and A.C dates in the timeline?  
Thank you.

**Comment by evanwill** [comment, marked as answer]
@cgrillo98 unfortunately, no--it isn't supported out of the box. 
The timeline is built using Liquid (which does not seem to support BC dates in date parsing) and only does very basic parsing of dates (manually, not using Liquid's limited date parsing abilities). The advantage of this is that it avoids a lot of errors that come up from trying to parse date fields (is forgiving of metadata inconsistency) and the timeline can be quickly reused for other types of visualizations (i.e. not dates)--the disadvantage is that it isn't very smart... 

That said, you could customize the timeline layout to work with BC dates--there are a few different approaches possible depending on your collection contents and aims. This might involve creating a new metadata field that can be used to correctly sort the items internally. For example, your "date" field might have a value like "300 BCE", with a "sort_date" field of "-300" (with year only). We would then tweak the Liquid logic in the timeline to sort using the "sort_date" values, and add logic to display "300 BCE" for negative numbers and "300 CE" for positive numbers.

**Reply by cgrillo98** [reply]
Thank you very much! more complex than expected. When you say "timeline layout" you mean Timeline.JS?

**Reply by evanwill** [reply]
@cgrillo98 I was referring to the main CollectionBuilder timeline layout which you would find in your project template in "_layouts/timeline.html". 
The TimelineJS project *does* support BCE dates by using negative years (i.e. `300 BCE` = `-300`)--CollectionBuilder has a built in option to [add a TimelineJS feature](https://collectionbuilder.github.io/cb-docs/docs/advanced/timelinejs/)--however, unfortunately because of the way we are parsing the dates in CollectionBuilder, your negative years won't come through in the data! 

If you have have an example repository, I could quickly show you how to customize it to get it working.

**Reply by cgrillo98** [reply]
Thank you, any help is welcome. 
My repository is https://github.com/cgrillo98/collectionbuilder-gh

**Reply by evanwill** [reply]
Just made a [PR on your repository](https://github.com/cgrillo98/collectionbuilder-gh/pull/1) implementing it!

**Reply by evanwill** [reply]
Notes for others who might be interested: unfortunately, proper numeric sorting can not be done with Liquid, see [CB-CSV issue](https://github.com/CollectionBuilder/collectionbuilder-csv/issues/43). If you want to sort BC dates, you need to use javascript instead. This is implemented in [CB-Sheets](https://github.com/CollectionBuilder/collectionbuilder-sheets), however, by default Sheets reproduces the same date parsing and sorting as the normal CB-GH Liquid-based version, so it doesn't support negative years to handle BC dates. The best option is to add a new "year" column, add BC dates as negative years (i.e. `300 BC ` = `-300`) then use those years for sorting the timeline. This means you will have to do date parsing in your metadata work setting up the "year" column, rather than doing it in the template logic. You can check out the [example implementation gist](https://gist.github.com/evanwill/abe532493a9a90b9a3db5172823e86f2).

**Reply by cgrillo98** [reply]
I have a new question. The card that displays the timeline access on the home and the timeline.js are not working correctly anymore, do you know why is it happening?

**Reply by evanwill** [reply]
@cgrillo98 for the TimelineJS feature, I just opened a [PR switching it from using the old "date" column to the new "year" column](https://github.com/cgrillo98/collectionbuilder-gh/pull/2), so that the negative years can be passed to TimelineJS. 

For the time box on the home page--there isn't a nice quick fix, because of the issues with Liquid sorting--so I think your best option is to just manually edit the box (or remove it). To edit the box, take a look at the file ["_includes/index/time.html](https://github.com/cgrillo98/collectionbuilder-gh/blob/main/_includes/index/time.html) in your project. You can delete all the Liquid (stuff inside `{%   %}`). Then where it says `{{ date-range | first }} a {{ date-range | last }}`, replace the values with your range (so like `1300 BC a 2023`).

**Reply by cgrillo98** [reply]
Ok, that´s perfect. Thank you very much again!! : )

**Reply by evanwill** [reply]
@cgrillo98 I think you "approved" [my recent pull request](https://github.com/cgrillo98/collectionbuilder-gh/pull/3), but didn't merge it. Here is the [github documentation for merging PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request). Let me know if that works!

**Reply by cgrillo98** [reply]
Yes! thx.
I can´t merge one of te pulls because there is a conflict on the timelinejs.json

**Reply by evanwill** [reply]
@cgrillo98 your site still isn't building correctly on GitHub Pages--> the issue is with using the rgb values in "_sass/_custom.scss". The way you have them written, like `background-color: rgb(0 0 0 / 31%) !important;` actually works in newer versions of SASS, but the version of SASS used by GitHub Pages is old! These values cause an error, so you site is not rebuilt.

**Reply by cgrillo98** [reply]
Ok,thank you. 
Now it works perfectly!
